corehq/apps/domain/forms.py

```diff
@@ -623,8 +623,8 @@ def _save_account_confirmation_settings(self, domain):
             settings.save()
 
     def _save_release_mode_setting(self, domain):
-        if self.cleaned_data.get("release_mode_visibility"):
-            setting_obj = AppReleaseModeSetting.get_settings(domain=domain.name)
+        setting_obj = AppReleaseModeSetting.get_settings(domain=domain.name)
+        if self.cleaned_data.get("release_mode_visibility") != setting_obj.is_visible:
```

Comment:
Can you share the reason for this change?

Reply:
To save object again only in case of change from existing.